src/Adapter/MSTest.CoreAdapter/Resources/Resource.resx

```diff
@@ -368,4 +368,19 @@ Error: {1}</value>
   <data name="OlderTFMVersionFoundClassCleanup" xml:space="preserve">
     <value>An older version of MSTestV2 package is loaded in assembly, test cleanup methods might not run as expected. Please make sure all your test projects references MSTest pacakges newer then version 2.2.8.</value>
   </data>
+  <data name="CannotExpandIDataSourceAttribute" xml:space="preserve">
+    <value>Exception occurred while expanding IDataSource rows from attribute on "{0}.{1}": {2}</value>
+    <comment>{0}: TypeName with namespace, 
+{1}: Method name,</comment>
+  </data>
+  <data name="CannotExpandIDataSourceAttribute_DuplicateDisplayName" xml:space="preserve">
+    <value>Display name "{2}" on indexes {0} and {1} are duplicate.</value>
+    <comment>{0}, {1}: Zero based index if an element inside of an array
+{2}: Display name of a test.</comment>
```

Comment:
```suggestion
    <comment>{0}: Zero-based index of the first data row with the given display name
{1}: Zero-based index of the duplicate data row display name
{2}: Test display name.</comment>
```

Reply:
Comments are not that important, it's to help loc team.